### DISCUSSION TITLE
Jail for only an IP range?

### DISCUSSION THREAD

**Discussion by robozb** [fail2ban/fail2ban#3287] [General]
Dear All,

I would like to create separate Jails for the local network and for the "outside world". So I need to specify IP ranges per Jails. 

I don't have problem with the "outside world" Jail because I can use ignoreip option there to exclude the local network.

But how can I specify the IP range for local network only Jail?

Thank you very much for your any help!

Regards: Beci

**Comment by sebres** [comment]
There is no inverse lookup at the moment in stock fail2ban for `ignoreip`, but...
1. one is able to use FQDN names in `ignoreip`, so if you have own DNS server you could specify some name for few subnets corresponding "outside world" to ignore them if you'd add them to `ignoreip`.
2. fail2ban has also an option `ignorecommand` so you can check whether the IP is in some subnet using that...  you could even cache the values to speed-up the lookup a bit. Here is an example using python and fail2ban API together with cache:
```ini
ignorecache = key="<ip>", max-count=1000, max-time=5m
ignorecommand = fail2ban-python -c 'import sys; from fail2ban.server.ipdns import IPAddr; sys.exit(1 if IPAddr(sys.argv[1]).isInNet(IPAddr("192.0.2.1/24")) else 0)' "<ip>"
```
3. split the logs/journals if possible (so both jails would simply monitor different logs and wouldn't need ignore check at all), see https://github.com/fail2ban/fail2ban/wiki/Best-practice#reduce-parasitic-log-traffic for example

**Comment by robozb** [comment]
Dear Sebres, Thank you so much again!!! I will try all of these! Best Regards: Bela
